client/src/partials/finance/cashflow/cashflow.js

```diff
@@ -0,0 +1,326 @@
+    }
+
+    /**
+     * @function labelization
+     * @description process for getting unique labels for incomes and expenses,
+     * and all totals needed
+     */
+    function labelization () {
+      var uniqueIncomes = [], uniqueExpenses = [];
+      session.incomesLabels = uniquelize(session.incomesLabels);
+      session.expensesLabels = uniquelize(session.expensesLabels);
+
+      /** incomes rows */
+      session.periodicData.forEach(function (flow) {
+        session.incomes[Dates.util.str(flow.period.start_date)] = {};
+        session.incomesLabels.forEach(function (label) {
+          session.summationIncome[Dates.util.str(flow.period.start_date)].forEach(function (transaction) {
+            if (transaction.transfer_type === label) {
+              session.incomes[Dates.util.str(flow.period.start_date)][label] = vm.onlyCashes ? transaction.value : Exchange.convertToEnterpriseCurrency(session.currency_id, new Date(), transaction.value);
+            }
+          });
+        });
+      });
+
+      /** totals incomes rows */
+      session.periodicData.forEach(function (flow) {
+        session.totalIncomes[Dates.util.str(flow.period.start_date)] = 0;
+        session.summationIncome[Dates.util.str(flow.period.start_date)].forEach(function (transaction) {
+          session.totalIncomes[Dates.util.str(flow.period.start_date)] += vm.onlyCashes ? transaction.value : Exchange.convertToEnterpriseCurrency(session.currency_id, new Date(), transaction.value);
+        });
+      });
+
+      /** expense rows */
+      session.periodicData.forEach(function (flow) {
+        session.expenses[Dates.util.str(flow.period.start_date)] = {};
+        session.expensesLabels.forEach(function (label) {
+          session.summationExpense[Dates.util.str(flow.period.start_date)].forEach(function (transaction) {
+            if (transaction.transfer_type === label) {
+              session.expenses[Dates.util.str(flow.period.start_date)][label] = vm.onlyCashes ? transaction.value : Exchange.convertToEnterpriseCurrency(session.currency_id, new Date(), transaction.value);
+            }
+          });
+        });
+      });
+
+      /** totals expenses rows */
+      session.periodicData.forEach(function (flow) {
+        session.totalExpenses[Dates.util.str(flow.period.start_date)] = 0;
+        session.summationExpense[Dates.util.str(flow.period.start_date)].forEach(function (transaction) {
+          session.totalExpenses[Dates.util.str(flow.period.start_date)] += vm.onlyCashes ? transaction.value : Exchange.convertToEnterpriseCurrency(session.currency_id, new Date(), transaction.value);
+        });
+      });
+
+    }
+
+    /**
+     * @function uniquelize
+     * @param {array} array An array in which we want to get only unique values
+     * @description return an array which contain only unique values
+     */
+    function uniquelize (array) {
```

Comment:
Here's a much cleaner way to make a unique array that avoids creating an intermediate object.
```js
function unique(array) {

  // the first param is our new array of unique values
  // the second param is the next value in the array.
  return array.reduce(function (uniq, value) {

    // if we haven't seen the value yet, add it to the array (otherwise, ignore it).
    if (uniq.indexOf(value) === -1) { uniq.push(value); }
    return uniq;

  }, []);  // initialize with an empty array
}
```